ktor-throughput-benchmark/scripts/profile_benchmark.py

```
@@ -0,0 +1,339 @@
+        "-e", event,
+        "-o", "collapsed",  # Text format: collapsed stacks
+        "-f", str(output_file),
+        str(pid)
+    ]
+
+    print(f"Running: {' '.join(cmd)}", file=sys.stderr)
+
+    try:
+        subprocess.run(cmd, check=True)
+    except subprocess.CalledProcessError as e:
+        raise RuntimeError(f"async-profiler failed: {e}")
```

Ruff (0.15.2)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)